### DISCUSSION TITLE
Add reference @export properties

### DISCUSSION THREAD

**Discussion by Furgl** [godotengine/godot-proposals#10869] [Scripting]
### Context
I'm creating a GameState system where each GameState is a custom resource. I would like each GameState to know which Scene to load when it becomes active and to also know which other GameState to switch to when the escape key is pressed. 

Example code:
```gdscript
class_name GameState extends Resource

@export var scene: PackedScene
@export var escape_state: GameState
```
### The Problem
**1. Resources that are properties of a resource seem to be auto-loaded when that resource is loaded**
In this case, `scene` is auto-loaded when the `GameState` is loaded. This is not ideal because I want all of the `GameState`s to be loaded, but only the currently active `GameState`'s `scene` to be loaded.

**2. Resource properties cannot create circular references**
During gameplay, I want pressing escape to open a pause menu. Then pressing escape again should close the pause menu and return to gameplay. 
To do that, I would create two `GameState`s:
- A Gameplay `GameState` with `escape_state` set to a Pause_Menu `GameState`
- A Pause_Menu `GameState` with `escape_state` set to the Gameplay `GameState`

This cannot be done and causes this error to appear:
> Recursion detected, unable to assign resource to property.
### Possible Solution
An `@export_ref` annotation for a `String` property that stores the path of the resource, but the benefit would be that the path updates if the resource is moved or renamed to allow for easy refactoring. At runtime, the resource can be loaded from the path as needed.
### Alternatives
`scene` and `escape_state` could be changed to `String` and set to the paths of their resources, but that would not support moving or renaming resources and would make refactoring difficult.

**Comment by pixel-boy** [comment]
I wonder if UIDs will make this easier to do, maybe it's something planned?

In any case the current limitation has often bothered me, and it would be nice to have an automatic update of the exported file paths.

**Comment by jitspoe** [comment]
Should this be a discussion or an issue?  I'm also running into an issue with resources being referenced and loaded as a resource, but with importer settings.  Storing a reference to a resource inside of the default preset for an importer causes the whole project to fail to load: https://github.com/godotengine/godot/issues/83316

Other related issues:
https://github.com/godotengine/godot/issues/99267
https://github.com/godotengine/godot-proposals/issues/8350

**Comment by rpahut** [comment]
There is a way to reference a resource without loading it - by its path. What was missing in the past is the ability to have these paths managed by the editor, as it would a direct reference. The currently existing @export_file seem to allow that. Only it won't show up when attempting to delete the referenced resource, so it isn't exactly complete.
